### ISSUE TITLE
i have some confused  in your code

### ISSUE THREAD

**Issue by daixiangzi** [iBelieveCJM/pseudo_label-pytorch#2]
labeled_bs = self.labeled_bs
labeled_loss = torch.sum(self.loss_fn(outputs, targets)) / labeled_bs
 with torch.no_grad():
           pseudo_labeled = outputs.max(1)[1]
unlabeled_loss = torch.sum(targets.eq(NO_LABEL).float() * self.loss_fn(outputs, pseudo_labeled)) / (data.size(0)-labeled_bs +1e-10)
loss = labeled_loss + self.unlabeled_weight()*unlabeled_loss
--------------------------------
 pseudo_labeled = outputs.max(1)[1] #outputs should be unlabel data,but your code outputs is  model(label_data)??????

**Comment by daixiangzi** [comment]
i seems know ,sorry,your code is right

**Comment by kkk142** [comment]
Have you duplicate the code?
